internal/services/springcloud/spring_cloud_active_deployment_resource.go

```diff
@@ -1,25 +1,25 @@
 package springcloud
 
 import (
+	"context"
 	"fmt"
 	"log"
 	"time"
 
-	"github.com/Azure/azure-sdk-for-go/services/preview/appplatform/mgmt/2021-09-01-preview/appplatform"
+	"github.com/Azure/azure-sdk-for-go/services/preview/appplatform/mgmt/2022-01-01-preview/appplatform"
 	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/parse"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/validate"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
-	"github.com/hashicorp/terraform-provider-azurerm/utils"
 )
 
 func resourceSpringCloudActiveDeployment() *pluginsdk.Resource {
 	return &pluginsdk.Resource{
-		Create: resourceSpringCloudActiveDeploymentCreate,
+		Create: resourceSpringCloudActiveDeploymentCreateUpdate,
 		Read:   resourceSpringCloudActiveDeploymentRead,
-		Update: resourceSpringCloudActiveDeploymentUpdate,
+		Update: resourceSpringCloudActiveDeploymentCreateUpdate,
```

Comment:
this should be using a combined Create and Update - can we revert this?

Reply:
Sorry I didn't get it. Can you elaborate on that?

Reply:
i think he meant "should not" as in revert combining create & update

Reply:
Got it, thanks!

Reply:
@katbyte , but the logic in Create and Update are almost the same, is it really necessary to revert combining create & update?

Reply:
yes these need to be separate - as the Create can assume all properties are set but the Update should only change properties if it's changed (e.g. using HasChanges)

Reply:
@tombuildsstuff , Got it. I've updated this PR, please take another look, thanks!